core/deployment/src/main/java/io/quarkus/deployment/steps/LocaleProcessor.java

```diff
@@ -139,13 +139,13 @@ public static String nativeImageIncludeLocales(NativeConfig nativeConfig, Locale
             return "all";
         }
 
-        // We subtract what we already declare for native-image's user.language or user.country.
+        // Ensure the default locale is included to ensure compatibility with GraalVM for JDK 24, which doesn't set the locale at build-time.
```

Comment:
```suggestion
        // GraalVM for JDK 24 doesn't include the default locale used at build time. We must explicitly include all
        // locales - including the build-time locale. This might include more locales than necessary for older GraalVM
        // versions, but should be mostly OK.
```

Reply:
> This might include more locales than necessary for older GraalVM versions, but should be mostly OK.
        
I think that's not true. The default locale is/was included by default with older versions as well (due to the static analysis considering it reachable AFAIU)

I updated the doc excluding this part.